core/app_modules/inventory.js

```diff
@@ -40,9 +40,52 @@ inventory.get = co.wrap(function*(call) {
   return yield Promise.resolve(results.rows[0]);
 });
 
+inventory.advancedSearch = co.wrap(function* (params) {//used for api from hlr.byu.edu
+  // Currently only supports single language and media type as hlr.byu only uses single option
+  var client = db();
+  var query = "SELECT inventory.call, title, quantity, is_reserve FROM inventory ";
+  var hasLanguage = false;
+  var hasMedia = false;
+  var lang, media;
+  if (params.language_code){
+    query+=" LEFT JOIN languages_items li on li.inventory_call = inventory.call " ;
+    lang = params.language_code;
+    delete params['language_code'];
+    hasLanguage = true;
+  }
+  if (params.media_items){
+    query+=" LEFT JOIN media_items mi on mi.call = inventory.call " ;
+    hasMedia = true;
+    media = params.media_items;
+    delete params['media_items'];
+  }
+  var len = Object.keys(params).length;
+  if (len>0){
+    query += " WHERE ";
+    var i = 0;
+    for (var item in params){
+      query += item + " ILIKE '%" + params[item] + "%'";
```

Comment:
There's a SQL query injection possible here; can you change this to use prepared statements?